client/signup/main.jsx

```diff
@@ -264,22 +265,23 @@ class Signup extends React.Component {
 			debug( 'From query string: vertical = %s', vertical );
 
 			const siteTopicStepName = 'site-topic';
+			const siteTopicSlug = toSiteTopicSlug( vertical );
```

Comment:
I believe `toSiteTopicSlug()` is not required here, and it should be okay to use `vertical` value directly. I think the english value is needed only in _step-actions.js_ just before sending the values to the _sites_new_ endpoint (check my comment below on the site-topic step for a more details).